### PR TITLE
Added feature to convert parameters to different units.

### LINKED ISSUE #7: Implement feature to convert parameters to different units
- [x] <b>Description</b>
    Add a utility to convert values from one unit to another: <br>
     - Temperature (Celsius, Kelvin, Fahrenheit)
     - Pressure
     - Volume (Gallon, Litres etc)
     - Currency (USD, GBP, CAD, INR)
     - Storage (Bytes, KB, MB, GB, TB, PB etc)
     - Distance
- [x] <b>Use cases</b>
    1. Convert values from one unit to another.
- [x] <b>Proposed design</b>
    For the currency converter, take the time instant for which the currency conversion is needed as well.
- [x] <b>Alternative design approaches</b>
    N/A.
- [x] <b>Workaround</b>
    N/A.
- [x] <b>Additional information</b>
    N/A.

### PR DESCRIPTION
**Pull request by the-c0d3br34k3r**
## Description
Closes #7.
Added the following parameters with support to convert to different units:
 - Currency
 - Data storage
 - Distance
 - Measurable
 - Pressure
 - Temperature
 - Volume
 
## Author's Checklist
- [x] **These changes don't break existing dependants.** If they do, and you suggest to proceed, 
please explain why.

## Reviewer's Checklist
- [x] **All classes have documentation comments.**
- [x] **All methods have documentation comments.**
- [x] **No parameters are hardcoded.** Should be loaded as properties instead. If there needs to be hardcoded properties, explain why.
- [x] **All information to be logged/displayed to the user are internationalized.** If not, explain why. 
- [x] **README.md has been updated if needed.**
- [x] **pom.xml version is set to the latest date** If not, explain why.

## Assignee's Checklist
- [x] **All GitHub action checks have passed.**
- [x] **All reviewers have approved.**
- [x] **Relevant documentation has been updated if needed.**
- [x] **pom.xml version is set to the latest date** If not, explain why.

### CODE REVIEW

File: src/main/java/org/padaiyal/utilities/aayvalar/parameters/CurrencyParameter.java [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+   * @param desiredCurrencyUnit Desired output currency unit.
+   * @return The currency value in the desired unit.
+   * @throws IOException If there is an issue while invoking the API.
+   */
+  public double convertTo(String desiredCurrencyUnit)
+      throws IOException {
+    // Input validation
+    if (desiredCurrencyUnit.length() != 3) {
+      throw new IllegalArgumentException("Invalid currency unit - " + desiredCurrencyUnit
+          + ".\nCurrency unit should be exactly 3 letters.");
```
Comment (silvs110) [inline src/main/java/org/padaiyal/utilities/aayvalar/parameters/CurrencyParameter.java]: Internationalization

File: src/main/java/org/padaiyal/utilities/aayvalar/parameters/CurrencyParameter.java [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+        new BufferedInputStream(
+            con.getInputStream()
+        ).readNBytes(100) // The desired response should be way lesser than 100 bytes.
+    );
+    ObjectMapper mapper = new ObjectMapper();
+    JsonNode actualObj = mapper.readTree(response);
+    logger.debug(response);
+    return actualObj.get(
+        String.format(
+            "%s_%s",
```
Comment (silvs110) [inline src/main/java/org/padaiyal/utilities/aayvalar/parameters/CurrencyParameter.java]: This should be in the properties file.

File: src/main/java/org/padaiyal/utilities/aayvalar/parameters/CurrencyParameter.java [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+            desiredCurrencyUnit,
+            apiKey
+        )
+    );
+    HttpURLConnection con = (HttpURLConnection) url.openConnection();
+    con.setRequestMethod("GET");
+    String response = new String(
+        new BufferedInputStream(
+            con.getInputStream()
+        ).readNBytes(100) // The desired response should be way lesser than 100 bytes.
```
Comment (silvs110) [inline src/main/java/org/padaiyal/utilities/aayvalar/parameters/CurrencyParameter.java]: This should be in the properties file.

File: src/test/java/org/padaiyal/utilities/aayvalar/parameters/CurrencyParameterTest.java [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+
+    Assertions.assertEquals(currentValue, currencyParameter.getValue());
+
+    Assertions.assertEquals(currentUnit, currencyParameter.getUnit());
+
+    double resultValue = currencyParameter.convertTo(resultUnit);
+    Assertions.assertTrue(resultValue > 0);
+  }
+
+  @ParameterizedTest
```
Comment (silvs110) [inline src/test/java/org/padaiyal/utilities/aayvalar/parameters/CurrencyParameterTest.java:68]: Missing docstring

File: src/test/java/org/padaiyal/utilities/aayvalar/parameters/DataStorageParameterTest.java [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.converter.ConvertWith;
+import org.junit.jupiter.params.provider.CsvSource;
+import org.padaiyal.utilities.aayvalar.parameters.units.DataStorageUnitEnum;
+
+/**
+ * Tests the DataStorageParameter.
+ */
+public class DataStorageParameterTest
```
Comment (silvs110) [inline src/test/java/org/padaiyal/utilities/aayvalar/parameters/DataStorageParameterTest.java:11]: Missing tests docstrings

File: src/test/java/org/padaiyal/utilities/aayvalar/parameters/CurrencyParameterTest.java [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+      "3, CAD, USD",
+      "2, USD, INR",
+      "3.14, INR, USD",
+      "4.5, CAD, INR",
+      "90, INR, CAD",
+      "10, GBP, USD",
+      "1000, USD, GBP",
+      "100000001, GBP, USD"
+  })
+  void testConvertToWithValidInputs(Double currentValue, String currentUnit, String resultUnit)
```
Comment (silvs110) [inline src/test/java/org/padaiyal/utilities/aayvalar/parameters/CurrencyParameterTest.java:44]: missing docstring

File: src/main/java/org/padaiyal/utilities/aayvalar/parameters/units/VolumeUnitEnum.java [hunk 7]
```diff
@@ -0,0 +1,35 @@
+package org.padaiyal.utilities.aayvalar.parameters.units;
+
+/**
+ * Abstracts the volume parameter units.
+ */
+public enum VolumeUnitEnum implements UnitTypeEnumInterface<Double> {
```
Comment (silvs110) [inline src/main/java/org/padaiyal/utilities/aayvalar/parameters/units/VolumeUnitEnum.java:6]: Missing docstrings

File: src/main/java/org/padaiyal/utilities/aayvalar/parameters/units/TemperatureUnitEnum.java [hunk 8]
```diff
@@ -0,0 +1,31 @@
+package org.padaiyal.utilities.aayvalar.parameters.units;
+
```
Comment (silvs110) [inline src/main/java/org/padaiyal/utilities/aayvalar/parameters/units/TemperatureUnitEnum.java:2]: Missing method docstrings

File: src/main/java/org/padaiyal/utilities/aayvalar/parameters/units/PressureUnitEnum.java [hunk 9]
```diff
@@ -0,0 +1,33 @@
+package org.padaiyal.utilities.aayvalar.parameters.units;
+
```
Comment (silvs110) [inline src/main/java/org/padaiyal/utilities/aayvalar/parameters/units/PressureUnitEnum.java:2]: missing method docstrings

File: src/main/java/org/padaiyal/utilities/aayvalar/parameters/units/DistanceUnitEnum.java [hunk 10]
```diff
@@ -0,0 +1,37 @@
+package org.padaiyal.utilities.aayvalar.parameters.units;
+
```
Comment (silvs110) [inline src/main/java/org/padaiyal/utilities/aayvalar/parameters/units/DistanceUnitEnum.java:2]: Missing method docstring

**Review by silvs110 — CHANGES_REQUESTED** [review]
Shouldn't overriden methods have their own docstring?

File: src/main/java/org/padaiyal/utilities/aayvalar/parameters/units/PressureUnitEnum.java (hunk 9 above)
Comment (the-c0d3br34k3r) [reply src/main/java/org/padaiyal/utilities/aayvalar/parameters/units/PressureUnitEnum.java:2]: Skipped as they are overridden methods.

File: src/main/java/org/padaiyal/utilities/aayvalar/parameters/units/TemperatureUnitEnum.java (hunk 8 above)
Comment (the-c0d3br34k3r) [reply src/main/java/org/padaiyal/utilities/aayvalar/parameters/units/TemperatureUnitEnum.java:2]: Skipped as they are overridden methods.

File: src/main/java/org/padaiyal/utilities/aayvalar/parameters/units/VolumeUnitEnum.java (hunk 7 above)
Comment (the-c0d3br34k3r) [reply src/main/java/org/padaiyal/utilities/aayvalar/parameters/units/VolumeUnitEnum.java:6]: Skipped as they are overridden methods.

File: src/test/java/org/padaiyal/utilities/aayvalar/parameters/DataStorageParameterTest.java (hunk 5 above)
Comment (the-c0d3br34k3r) [reply src/test/java/org/padaiyal/utilities/aayvalar/parameters/DataStorageParameterTest.java:11]: Skipped as they are overridden methods.

File: src/main/java/org/padaiyal/utilities/aayvalar/parameters/CurrencyParameter.java (hunk 1 above)
Comment (the-c0d3br34k3r) [reply src/main/java/org/padaiyal/utilities/aayvalar/parameters/CurrencyParameter.java]: Done.

File: src/main/java/org/padaiyal/utilities/aayvalar/parameters/CurrencyParameter.java (hunk 2 above)
Comment (the-c0d3br34k3r) [reply src/main/java/org/padaiyal/utilities/aayvalar/parameters/CurrencyParameter.java]: Done.

File: src/main/java/org/padaiyal/utilities/aayvalar/parameters/CurrencyParameter.java (hunk 3 above)
Comment (the-c0d3br34k3r) [reply src/main/java/org/padaiyal/utilities/aayvalar/parameters/CurrencyParameter.java]: Done.

File: src/main/java/org/padaiyal/utilities/aayvalar/parameters/units/DistanceUnitEnum.java (hunk 10 above)
Comment (the-c0d3br34k3r) [reply src/main/java/org/padaiyal/utilities/aayvalar/parameters/units/DistanceUnitEnum.java:2]: Skipped as they are overridden methods.

File: src/test/java/org/padaiyal/utilities/aayvalar/parameters/CurrencyParameterTest.java (hunk 6 above)
Comment (the-c0d3br34k3r) [reply src/test/java/org/padaiyal/utilities/aayvalar/parameters/CurrencyParameterTest.java:44]: Done.

File: src/test/java/org/padaiyal/utilities/aayvalar/parameters/CurrencyParameterTest.java (hunk 4 above)
Comment (the-c0d3br34k3r) [reply src/test/java/org/padaiyal/utilities/aayvalar/parameters/CurrencyParameterTest.java:68]: Done.

**Review by silvs110 — APPROVED** [review]

**Pull request merged**